src/main/java/model/Faculty.java

```diff
@@ -1,16 +1,18 @@
 package model;
 
 
-import jakarta.persistence.Entity;
-import jakarta.persistence.GeneratedValue;
-import jakarta.persistence.Id;
+import jakarta.persistence.*;
 
+import java.util.Collection;
 import java.util.Objects;
 @Entity
 public class Faculty {
     @Id
     @GeneratedValue
     private long id;
+    @ManyToOne
+    @JoinColumn(name = "faculty_id")
```

Comment:
Тут должна быть вот такая связь связь:
```suggestion
    @OneToMany(mappedBy = "faculty")

```